diff --git a/common.mak b/common.mak
index 20b7fa3..2851b33 100644
--- a/common.mak
+++ b/common.mak
@@ -37,7 +37,8 @@ CPPFLAGS   := $(IFLAGS) $(CPPFLAGS)
 CFLAGS     += $(ECFLAGS)
 CCFLAGS     = $(CPPFLAGS) $(CFLAGS)
 ASFLAGS    := $(CPPFLAGS) $(ASFLAGS)
-CXXFLAGS   += $(CPPFLAGS) $(CFLAGS)
+STDC99FLAG := -std=c99
+CXXFLAGS   += $(CPPFLAGS) $(filter-out $(STDC99FLAG),$(CFLAGS))
 YASMFLAGS  += $(IFLAGS:%=%/) -Pconfig.asm

 HOSTCCFLAGS = $(IFLAGS) $(HOSTCPPFLAGS) $(HOSTCFLAGS)
